### PR TITLE
HOSTEDCP-1850: HyperShift operator: Allow setting Kube APIServer maximum requests inflight

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Adds support to the HyperShift operator for overriding default maximum requests in flight for Kubernetes API server.
Adds fields to the ClusterSizingConfiguration CRD that allows setting values for these based on cluster size.

This is the hypershift operator half of https://github.com/openshift/hypershift/pull/4488

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1850](https://issues.redhat.com/browse/HOSTEDCP-1850)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go [hunk 1]
```diff
@@ -771,6 +771,12 @@ func (r *DedicatedServingComponentSchedulerAndSizer) updateHostedCluster(ctx con
 			delete(hc.Annotations, hyperv1.MachineHealthCheckTimeoutAnnotation)
 		}
 	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MaximumRequestsInflight != nil {
+		hc.Annotations[hyperv1.KubeAPIServerMaximumRequestsInFlight] = fmt.Sprint(*sizeConfig.Effects.MaximumRequestsInflight)
```
Comment (jparrill) [inline hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go:775]: It's there any kind of validation around the annotation content?

**Review by jparrill — COMMENTED** [review]
Small question, otherwise lgtm

File: hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go (hunk 1 above)
Comment (csrwng) [reply hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go:775]: The CR ensures that it is a number. As far as limiting the values for that number, I'm not sure that we know just yet what those limits should be. This is an API that is meant to be consumed by the service provider (ie. fleet manager), so I'm less worried about validation here.

**Pull request merged**